query/src/main/java/org/infinispan/query/concurrent/InfinispanIndexingExecutorProvider.java

```diff
@@ -0,0 +1,70 @@
+package org.infinispan.query.concurrent;
+
+import static org.infinispan.factories.KnownComponentNames.BLOCKING_EXECUTOR;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.hibernate.search.backend.lucene.work.spi.LuceneWorkExecutorProvider;
+import org.hibernate.search.engine.common.execution.spi.SimpleScheduledExecutor;
+import org.infinispan.factories.AbstractComponentFactory;
+import org.infinispan.factories.AutoInstantiableFactory;
+import org.infinispan.factories.KnownComponentNames;
+import org.infinispan.factories.annotations.ComponentName;
+import org.infinispan.factories.annotations.DefaultFactoryFor;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+
+@Scope(Scopes.GLOBAL)
+@DefaultFactoryFor(classes = LuceneWorkExecutorProvider.class)
+public class InfinispanIndexingExecutorProvider extends AbstractComponentFactory implements AutoInstantiableFactory, LuceneWorkExecutorProvider {
+
+   @Inject
+   @ComponentName(KnownComponentNames.TIMEOUT_SCHEDULE_EXECUTOR)
+   ScheduledExecutorService scheduledExecutorService;
+
+   @Inject
+   @ComponentName(BLOCKING_EXECUTOR)
+   ExecutorService blockingExecutor;
```

Comment:
I would still suggest using BlockingManager#asExecutor so we can get tracing to know when invoking lucene indexing

Reply:
Also my final goal is so that no code uses the BLOCKING_EXECUTOR directly in general.

Reply:
Did it. I just pushed it

Reply:
> I need something that returns a CompletionStage or a Future when I submit (execute) the operation I think.

[CompletableFuture.](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#runAsync-java.lang.Runnable-java.util.concurrent.Executor-)
and supplyAsync for ones that need a value.

It looks like you did it a different way though, which is fine 👍